src/macros.jl

```diff
@@ -359,23 +368,42 @@ macro wrap(ex)
 
     fullargs = ex.args[2:length(ex.args)]
 
-    # strip type annotation from arguments for use in the call to GAP
-    args = [x isa Symbol ? x : x.args[1] for x in fullargs]
-
+    # splits the arguments with type annotations into expressions for the lhs and rhs
+    # of the call of the form `func(args) = GAP.Globals.func(args)`:
+    # - type annotations have to be removed for the rhs
+    # - if the type annotation is `GapObj`, the rhs has to be wrapped in `GAP.GapObj(...)` and the lhs has no type annotation
+    tempargs = [
+        begin
+            if x isa Symbol
+                (x, x)
+            elseif x.head == :(::) && length(x.args) == 2
+                var = x.args[1]
+                typeannot = x.args[2]
+                if typeannot in [:GapObj, :(GAP.GapObj)]
+                    (var, :(GAP.GapObj($var)::GAP.GapObj))
```

Comment:
```suggestion
                if typeannot in [:GapObj, :(GAP.GapObj)]
                    (var, :(GAP.GapObj($var)::GAP.GapObj))
                elseif typeannot == :(GAP.Obj)
                    (var, :(GAP.Obj($var)::GAP.Obj))
                elseif typeannot in [:GapInt, :(GAP.GapInt)]
                    (var, :(GAP.GapInt($var)::GAP.GapInt))
```
as suggested in https://github.com/oscar-system/GAP.jl/pull/991#discussion_r1603432546